docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,552 @@
+    -   `create_error`: Text description of the failure (e.g., "Insufficient cycles attached", "No subnets in which to create a canister", or errors from the IC management canister).
+    -   `refund_amount`: The amount of cycles refunded to the caller. If the request failed due to issues *before* attempting creation on a subnet (e.g., bad arguments, insufficient cycles attached initially), the full attached amount is usually refunded. If it failed *during* the cross-net call to a subnet (e.g., subnet rejected creation), a penalty (`1,000,000,000` cycles, `BAD_REQUEST_CYCLES_PENALTY`) is deducted before refunding the rest.
+
+#### Notes
+
+-   This method requires cycles to be attached to the call.
+-   Returns the principal of the newly created canister.
+-   The result is **not idempotent** — calling it again will consume cycles and create a different canister.
+
+## `get_icp_xdr_conversion_rate`
+
+Returns the current ICP/XDR exchange rate used to compute how many cycles a given amount of ICP buys.
+
+#### Parameters
+
+```candid
+record {}
+```
+
+#### Returns
+
+```candid
+record {
+  data: record { // Note: In current implementation, this is never null if the call succeeds.
+    xdr_permyriad_per_icp: nat64;
+    timestamp_seconds: nat64;
+  };
+  certificate: blob;
+  hash_tree: blob; // CBOR encoded hash tree witness
+}
+
+```
+
+-   `xdr_permyriad_per_icp`: The exchange rate, in 1/10000ths of an XDR per ICP (Permyriad). For example, 12500 means 1 ICP = 1.25 XDR.
+-   `timestamp_seconds`: Time the rate was recorded, in seconds since the Unix epoch.
+-   `certificate`: The IC data certificate authenticating the response.
+-   `hash_tree`: The CBOR-encoded hash tree proving the inclusion of the `data` in the certificate.
+
+#### Notes
+
+This method can be called as a query. Clients should verify the certificate and hash_tree against the root subnet's public key to ensure authenticity.
+
+## Subnet Configuration (Governance Methods)
+
+The following methods are used by the NNS Governance canister to configure how canisters are created across different subnets. Developers typically do not call these directly but may query the results using methods like `get_default_subnets`, `get_principals_authorized_to_create_canisters_to_subnets`, and `get_subnet_types_to_subnets`.
+
+### `set_authorized_subnetwork_list` (Governance only)
+
+Configures which principals are authorized to create canisters on specific lists of subnets.
+
+-   **Caller**: Must be the NNS Governance canister.
+-   **Purpose**:
+    -   Sets the **default list** of subnets where *any* principal can create canisters (when `who` parameter is `null`).
+    -   Sets a specific list of subnets where a designated `who` principal is authorized to create canisters. If the `subnets` list is empty, the authorization for `who` is removed.
+-   **Preconditions**: Subnets added to the default or authorized lists cannot already be assigned to a named subnet type (see `update_subnet_type`).
```

Comment:
is a "named subnet" the same as a "subnet type"? or do we mean sth else where?

Reply:
I guess subnet type is accurate